### DISCUSSION TITLE
Adaptive Bed Meshing is not working

### DISCUSSION THREAD

**Discussion by LouCipher20** [Guilouz/Creality-Helper-Script-Wiki#441] [Issues]
Hello,
It' seems the adaptive meshing is not working. I tried it with different models. It's always meshing the whole bed.
Updating KAMP comes within updates for the script helper?

**Comment by Guilouz** [comment]
Have you updated script ?

**Reply by LouCipher20** [reply]
Yes. Factory Reset, Firmware Update and installation of the helper script. Everything is working fine.
Even the Kamp purge line is working fine.

**Reply by rmasci** [reply]
I did that. It does a Kamp purge line but does the whole bed leveing thing every time.

**Comment by Guilouz** [comment]
And what the console said?

**Reply by LouCipher20** [reply]
`13:27:43 
$ BED_MESH_CALIBRATE
13:27:43 
// No object detected! Make sure you have enabled Exclude Objets setting in your slicer. Using Full Bed Mesh.
13:27:47 
// Generating new points...
13:27:47 
// bed_mesh: generated points
// Index | Tool Adjusted | Probe
13:27:47 
// 0 | (5.0, 5.0) | (5.0, 5.0)
13:27:47 
// 1 | (57.5, 5.0) | (57.5, 5.0)
13:27:48 
// 2 | (110.0, 5.0) | (110.0, 5.0)
.........
13:38:29 
// probe at 215.000,215.000 is z=-0.070281
13:38:29 
// Mesh Bed Leveling Complete
13:38:29 
// Bed Mesh state has been saved to profile [default]
// for the current session. The SAVE_CONFIG command will
// update the printer config file and restart the printer.
`

after starting a print:
`13:42:11 
File selected
13:42:14 
// wait temp end
13:42:14 
// x_axes: xyz
13:42:14 
// x_axes: xyz
13:42:29 
// x_park = -104.5
13:42:29 
// y_park = 104.0
13:42:47 
// Run Current: 0.56A Hold Current: 0.56A
13:44:51 
// Starting KAMP Bed Mesh...
13:44:52 
// Algorithm: lagrange
13:44:52 
// Default probe count: 5,5
13:44:52 
// Adapted probe count: 3,3
13:44:52 
// Default mesh bounds: 5.0,5.0, 215.0,215.0
13:44:52 
// Mesh margin is 0, margin not increased.
13:44:52 
// Fuzz amount is 0, mesh points not fuzzed.
13:44:52 
// Adapted mesh bounds: 90.1123,81.3695, 129.888,138.63
13:44:52 
// KAMP adjustments successful. Happy KAMPing!
13:44:52 
// Generating new points...
13:44:52 
// bed_mesh: generated points
// Index | Tool Adjusted | Probe
13:44:52 
// 0 | (90.1, 81.4) | (90.1, 81.4)
13:44:52 
// 1 | (110.0, 81.4) | (110.0, 81.4)
13:44:52 
// 2 | (129.9, 81.4) | (129.9, 81.4)
13:44:52 
// 3 | (129.9, 110.0) | (129.9, 110.0)
13:44:52 
// 4 | (110.0, 110.0) | (110.0, 110.0)
13:44:53 
// 5 | (90.1, 110.0) | (90.1, 110.0)
13:44:53 
// 6 | (90.1, 138.6) | (90.1, 138.6)
13:44:53 
// 7 | (110.0, 138.6) | (110.0, 138.6)
06:38:41 
// Running Command {v4l2-ctl}...:
13:44:53 
// 8 | (129.9, 138.6) | (129.9, 138.6)
06:38:41 
// Command {v4l2-ctl} finished
13:45:31 
// probe at 90.112,81.370 is z=0.132990
06:38:41 
// Running Command {v4l2-ctl}...:
13:45:36 
// probe at 109.992,81.370 is z=0.032844
06:38:41 
// unknown control 'focus_auto'
13:45:41 
// probe at 129.872,81.370 is z=0.016406
06:38:41 
// Command {v4l2-ctl} finished
13:45:48 
// probe at 129.872,109.999 is z=-0.063813
06:47:56 
$ BED_MESH_CALIBRATE
13:45:54 
// probe at 109.992,109.999 is z=-0.045063
06:47:57 
// No object detected! Make sure you have enabled Exclude Objets setting in your slicer. Using Full Bed Mesh.
13:45:59 
// probe at 90.112,109.999 is z=0.004687
06:48:00 
// Generating new points...
13:46:06 
// probe at 90.112,138.630 is z=-0.076531
06:48:00 
// bed_mesh: generated points
// Index | Tool Adjusted | Probe
13:46:11 
// probe at 109.992,138.630 is z=-0.112344
06:48:00 
// 0 | (5.0, 5.0) | (5.0, 5.0)
13:46:23 
// probe at 129.872,138.630 is z=-0.119595
13:46:24 
// Mesh Bed Leveling Complete
13:46:24 
// Bed Mesh state has been saved to profile [default]
// for the current session. The SAVE_CONFIG command will
// update the printer config file and restart the printer.
13:46:24 
// Bed Mesh state has been saved to profile [kamp]
// for the current session. The SAVE_CONFIG command will
// update the printer config file and restart the printer.
13:46:24 
// Smart Park location: 85.00015,71.3695
13:46:52 
// Starting KAMP line purge...
13:46:52 
// Moving filament tip 0.0mm
13:46:52 
// KAMP line purge starting at 85.00015, 71.3695 and purging 50.0mm of filament, requested flow rate is 12.0mm3/s.`

**Comment by Guilouz** [comment]
Looks the first line :  // No object detected! Make sure you have enabled Exclude Objets setting in your slicer. Using Full Bed Mesh.

That's means there is no object in your gcode.

**Reply by LouCipher20** [reply]
Ok. But the "Exclude Objects" Box is checked. I use the newest Beta 2.0.0 Orcaslicer.
Now I tried Creality Print. It was aqays working. But now, same hier. No objects.

This is a part from my gcode file. It looks ok?!

> ; EXECUTABLE_BLOCK_START
EXCLUDE_OBJECT_DEFINE NAME=Skull.stl_id_0_copy_0 CENTER=110,110 POLYGON=[[90.1123,104.835],[90.1138,103.796],[90.3657,98.725],[90.4183,97.958],[90.4507,97.5475],[90.5382,97.2115],[90.5677,97.101],[90.9528,96.11],[91.1217,95.693],[91.2098,95.566],[99.8372,83.7195],[99.9663,83.572],[100.084,83.464],[100.352,83.237],[100.465,83.1635],[100.693,83.051],[101.971,82.495],[102.235,82.386],[102.538,82.261],[102.762,82.169],[103.026,82.1135],[109.145,81.471],[109.813,81.4105],[110.569,81.3695],[111.132,81.4845],[111.361,81.5685],[111.572,81.71],[111.835,81.893],[123.163,90.991],[123.729,91.659],[123.863,91.836],[124.097,92.254],[124.218,92.4805],[124.338,92.803],[124.597,93.6405],[125.808,98.188],[127.447,104.764],[129.887,117.266],[129.888,117.488],[129.758,121.572],[129.718,122.177],[129.356,124.56],[129.29,124.874],[129.219,125.21],[128.449,127.663],[128.347,127.923],[128.241,128.131],[128.083,128.443],[127.872,128.804],[126.87,130.215],[126.764,130.338],[124.865,132.541],[124.443,133.023],[124.36,133.112],[123.441,134.067],[121.174,136.057],[120.961,136.203],[120.748,136.316],[117.585,137.744],[117.48,137.785],[116.779,138.042],[116.291,138.194],[115.724,138.368],[115.423,138.431],[112.046,138.621],[111.097,138.63],[109.936,138.623],[106.296,138.492],[105.98,138.459],[104.925,138.278],[104.239,138.128],[104.028,138.076],[100.721,136.914],[100.441,136.797],[100.198,136.655],[99.9532,136.507],[99.7867,136.397],[99.4082,136.093],[96.4317,133.526],[96.0627,133.185],[96.0097,133.135],[94.3217,131.188],[93.1088,129.659],[92.5282,128.873],[92.4227,128.682],[91.9642,127.854],[91.7943,127.48],[91.4732,126.674],[91.3677,126.387],[91.3208,126.213],[90.5767,122.563],[90.5347,122.253],[90.5238,122.115],[90.1123,104.835]]
M73 P0 R100
M106 S0
M106 P2 S0
;TYPE:Custom
SET_PRINT_STATS_INFO TOTAL_LAYER=352
M140 S0
M104 S0
START_PRINT EXTRUDER_TEMP=210 BED_TEMP=60

**Reply by lvlacleod** [reply]
I figured my issue out. I have knomi installed and in KNOMI.cfg, BED_MESH_CALIBRATE has a rename_existing line. I basically uncommented the entire BED_MESH_CALIBRATE macro and KAMP is working again. I'll let the print finish before I enable the BED_MESH_CALIBRATE macro and fix the issue

**Reply by kodishtv** [reply]
I did a factory reset on 3 of my printers, re-installed the helper scripts, creality prints Exclude Objects doesn't work:

15:05:15 
// No object detected! Make sure you have enabled Exclude Objets setting in your slicer. Using Full Bed Mesh.


 I even updated to the latest Creality Print version: V4.3.4.5553

I'll use a different slicer.

**Reply by Redmarlboroman** [reply]
> Looks the first line : // No object detected! Make sure you have enabled Exclude Objets setting in your slicer. Using Full Bed Mesh.
> 
> That's means there is no object in your gcode.

I am using creality print V6.3.0.3420, I also encounter this problem and in the "Others" tab, I have tick the exclude objects but still it does the full bed meshing as it always says no object. Please help on this problem. Thanks

**Comment by lvlacleod** [comment]
I've got the same problem. Exclude object is set to on but it always does a full bed mesh. If I turn off KAMP and BED LEVELING in fluidd it saves the mesh to kamp and default profiles. Only started happening since 5.0.0

I've not been able to check if KAMP is working now, since I discovered this. In gcode_macro.cfg after I uninstalled and re-installed KAMP about 50% of the macros were all commented out. I can assure you there is absolutely no way I did this myself as I literally did a fresh install of 1.3.3.8, factory reset and then installed all your scripts that I use.

**Comment by TrueBlueRoo** [comment]
Looks like Creality has disabled it or put a hook somewhere.

I tried the foolowing in Adaptive_Meshing.cfg:

  #Original entry
    _BED_MESH_CALIBRATE mesh_min={adapted_x_min},{adapted_y_min} mesh_max={adapted_x_max},{adapted_y_max} ALGORITHM={algorithm} PROBE_COUNT={points_x},{points_y}

# None of them below worked

    # BED_MESH_CALIBRATE AREA_START={adapted_x_min},{adapted_y_min} AREA_END={adapted_x_max},{adapted_y_max} ALGORITHM={algorithm} PROBE_COUNT={points_x},{points_y}
    # RUN_SHELL_COMMAND CMD="BED_MESH_CALIBRATE_AREA_START={adapted_x_min},{adapted_y_min} AREA_END={adapted_x_max},{adapted_y_max}"
    # BED_MESH_CALIBRATE_NATIVE XMIN={adapted_x_min} YMIN={adapted_y_min} XMAX={adapted_x_max} YMAX={adapted_y_max}

**Comment by skydivingwill** [comment]
I ran into this issue while trying this out this week. The guild isn't clear on how to trigger the Adaptive Bed Meshing. If you try sending and printing using Creality Api then it won't work. If you send only from Creality Print or Orca Slicer using Creality port and then print from Fluidd (probably Mainsail too) it will work. 

What I ultimately discovered is the you need to change host type in OrcaSlicer. I changed it from Creality to Octo/Klipper and added the 4408 port to ip address. This fixed the Adaptive Bed Meshing plus the device tab now shows the Fluidd web page.

[https://guilouz.github.io/Creality-Helper-Script-Wiki/slicers/orcaslicer/](https://guilouz.github.io/Creality-Helper-Script-Wiki/slicers/orcaslicer/)

If someone finds a way to get Creality Print to post solution.

**Reply by TrueBlueRoo** [reply]
You link isn't working. Tou have to remove the ?h=orcaslicer.
[https://guilouz.github.io/Creality-Helper-Script-Wiki/slicers/orcaslicer/](https://guilouz.github.io/Creality-Helper-Script-Wiki/slicers/orcaslicer/)
